homeassistant/components/amcrest/config_flow.py

```diff
@@ -0,0 +1,197 @@
+"""Config flow to configure Amcrest integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from amcrest import AmcrestError, LoginError
+import voluptuous as vol
+
+from homeassistant.config_entries import (
+    ConfigEntry,
+    ConfigFlow,
+    OptionsFlowWithConfigEntry,
+)
+from homeassistant.const import (
+    CONF_BINARY_SENSORS,
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SENSORS,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from . import AmcrestChecker
+from .binary_sensor import (
+    AUDIO_DETECTED_KEY,
+    AUDIO_DETECTED_POLLED_KEY,
+    BINARY_SENSORS,
+    CROSSLINE_DETECTED_KEY,
+    CROSSLINE_DETECTED_POLLED_KEY,
+    MOTION_DETECTED_KEY,
+    MOTION_DETECTED_POLLED_KEY,
+)
+from .const import (
+    CONF_CONTROL_LIGHT,
+    CONF_FFMPEG_ARGUMENTS,
+    CONF_RESOLUTION,
+    CONF_STREAM_SOURCE,
+    DEFAULT_CONTROL_LIGHT,
+    DEFAULT_FFMPEG_ARGUMENTS,
+    DEFAULT_NAME,
+    DEFAULT_PORT,
+    DEFAULT_RESOLUTION,
+    DEFAULT_STREAM_SOURCE,
+    DOMAIN,
+    RESOLUTION_LIST,
+    STREAM_SOURCE_LIST,
+)
+from .sensor import SENSORS
+
+_LOGGER = logging.getLogger(__name__)
+
+SETUP_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default=DEFAULT_NAME): str,
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    }
+)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_FFMPEG_ARGUMENTS, default=DEFAULT_FFMPEG_ARGUMENTS): str,
+        vol.Optional(CONF_RESOLUTION, default=DEFAULT_RESOLUTION): vol.In(
+            RESOLUTION_LIST
+        ),
+        vol.Optional(CONF_STREAM_SOURCE, default=DEFAULT_STREAM_SOURCE): vol.In(
+            STREAM_SOURCE_LIST
+        ),
+        vol.Optional(CONF_BINARY_SENSORS, default=[]): cv.multi_select(
+            {sensor.key: sensor.name for sensor in BINARY_SENSORS}
+        ),
+        vol.Optional(CONF_SENSORS, default=[]): cv.multi_select(
+            {sensor.key: sensor.name for sensor in SENSORS}
+        ),
+        vol.Optional(CONF_CONTROL_LIGHT, default=DEFAULT_CONTROL_LIGHT): bool,
```

Comment:
We should not allow them to select entities in the options flow. We should use the standard functionality of HA to enable/disable entities.

Instead set `entity_registery_enabled_default=True` using the old yaml so when they migrate, entities are there but disabled.